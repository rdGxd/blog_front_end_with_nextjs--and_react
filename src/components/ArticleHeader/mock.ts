import { ArticleHeaderProps } from ".";
import { data } from "../../api/dados.json";

const { title, excerpt, cover, author, categories, createdAt } = data.posts.data[0].attributes;

const mockArticleHeader = {
  title,
  author,
  categories,
  cover,
  createdAt,
  excerpt,
} as ArticleHeaderProps;

export default mockArticleHeader;
