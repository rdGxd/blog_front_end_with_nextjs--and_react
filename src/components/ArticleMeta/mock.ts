import { ArticleMetaProps } from ".";
import { data } from "../../api/dados.json";

const { categories, author, createdAt } = data.posts.data[0].attributes;

const mockArticleMeta = {
  createdAt,
  categories: categories.data,
  author,
} as ArticleMetaProps;

export default mockArticleMeta;
