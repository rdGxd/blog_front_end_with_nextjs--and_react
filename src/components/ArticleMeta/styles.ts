import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ", ";
    }

    .categories span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(50%);
    }
  `}
`;
