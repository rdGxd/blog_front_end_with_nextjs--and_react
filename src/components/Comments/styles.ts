import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
