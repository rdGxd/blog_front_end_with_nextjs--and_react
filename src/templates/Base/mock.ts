import { BaseProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting.data,
  children: "Ol√° mundo",
} as unknown as BaseProps;
