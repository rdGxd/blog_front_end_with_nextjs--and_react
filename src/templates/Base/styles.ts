import styled, { css } from "styled-components";

export const Wrapper = styled.div``;

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    ${theme.spacings.xhuge}
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
