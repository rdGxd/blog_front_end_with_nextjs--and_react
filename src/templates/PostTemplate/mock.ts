import { PostTemplateProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting,
  post: data.posts.data[0].attributes,
} as unknown as PostTemplateProps;
