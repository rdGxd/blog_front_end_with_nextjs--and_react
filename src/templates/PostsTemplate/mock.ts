import { PostsTemplateProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting,
  posts: { data: data.posts.data },
} as unknown as PostsTemplateProps;
