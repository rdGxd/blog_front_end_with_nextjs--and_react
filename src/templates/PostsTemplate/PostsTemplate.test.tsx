import { PostsTemplate } from ".";
import { renderTheme } from "../../styles/render-theme";
import mock from "./mock";

describe("<PostsTemplate />", () => {
  it("should match snapshot", () => {
    const { container } = renderTheme(<PostsTemplate {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
