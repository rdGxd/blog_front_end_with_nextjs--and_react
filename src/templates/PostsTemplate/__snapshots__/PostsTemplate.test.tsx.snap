// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PostsTemplate /> > should match snapshot 1`] = `
.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c16 .c5 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c16 a {
  color: inherit;
}

.c16:hover a {
  color: #dc143c;
}

.c16:hover img {
  opacity: 0.8;
}

.c17 {
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
}

.c14 {
  width: 100%;
}

.c15 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c21 {
  font-size: 2.4rem;
}

.c21 p {
  margin: 2.4rem 0;
}

.c21 a:visited,
.c21 a:link {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c21 a:hover {
  filter: brightness(80%);
}

.c21 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c21 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c21 pre code {
  color: inherit;
  background: inherit;
}

.c21 img {
  max-width: 100%;
}

.c21 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c21 .image figcaption {
  text-align: center;
  font-size: 1.6rem;
  padding: 1.6rem;
  line-height: 1.3;
}

.c21 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c21 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c21 ul,
.c21 ol {
  margin: 2.4rem 4.0rem;
}

.c21 .table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c21 table {
  width: 100%;
  border-collapse: collapse;
}

.c21 table td,
.c21 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c21 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c19 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c19 a {
  color: inherit;
  text-decoration: none;
}

.c19 .c20 {
  font-size: 1.6rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c22 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
}

.c7 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c7 img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c10 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem -1rem);
}

.c11 {
  display: flex;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c8 {
  display: block;
  color: #FFFFFF;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  display: flex;
  background: #000000;
  padding: 3.2rem;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -35rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: flex;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.0rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  left: 1rem;
  color: #FFFFFF;
}

.c9 {
  margin-top: 6.4rem;
  padding: 0 3.2rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c18 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width: 360px) {
  .c15 {
    grid-template-columns: 1fr;
  }
}

@media (max-width: 768px) {
  .c21 {
    font-size: 2rem;
  }

  .c21 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width: 360px) {
  .c10 {
    display: flex;
    flex-flow: column wrap;
  }

  .c10>.c5 {
    margin: 0 0 2.4rem 0;
  }
}

@media (max-width: 360px) {
  .c11 {
    margin-left: 0;
  }
}

@media (max-width: 768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width: 768px) {
  .c0 {
    left: -0.1rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Rodrigo Silva"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="TESTE"
          target="_self"
        >
          #TESTE
        </a>
      </nav>
    </div>
    <header
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Rodrigo Silva - Blog do Rodrigo"
              src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            Rodrigo Silva
          </h2>
          <p>
            Blog do Rodrigo
          </p>
        </div>
      </header>
    </header>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <a
              href="/post/6-papapa"
            >
              <img
                alt="6PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/360_F_571067620_JS_5_T5_Tk_Dtu3gf8_Wqm78_Ko_JRF_1vob_Pvo6_1dff1a52c6.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/6-papapa"
              >
                6PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/7-papapa"
            >
              <img
                alt="7PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/bf2191273cf0b9c83d0ee39819fe02b2_0b947ee9d4.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/7-papapa"
              >
                7PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/8-papapa"
            >
              <img
                alt="8PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490198/linhas_brilhantes_na_forma_3d_do_coracao_humano_em_fundo_escuro_generativo_ai_191095_1435_071d182d7c.webp"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/8-papapa"
              >
                8PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/9-papapa"
            >
              <img
                alt="9PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/ea8003eda835af024292148060887ed4_5f4ba76b6a.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/9-papapa"
              >
                9PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/2-papapa"
            >
              <img
                alt="2PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489865/1213_78bc74760f.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/2-papapa"
              >
                2PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum. Pellentesque convallis hendrerit est a auctor.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/10-papapa"
            >
              <img
                alt="10PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490198/baixados_1_a01f56f752.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/10-papapa"
              >
                10PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/3-papapa"
            >
              <img
                alt="3PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/3-papapa"
              >
                3PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum. Pellentesque convallis hendrerit est a auctor.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/4-papapa"
            >
              <img
                alt="4PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490202/uma_pintura_de_um_lago_de_montanha_com_uma_montanha_ao_fundo_188544_9126_bc5fa577b5.webp"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/4-papapa"
              >
                4PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/5-papa-pa"
            >
              <img
                alt="5PAPAPa"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490200/pexels_pixabay_36717_154aaae2b0.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/5-papa-pa"
              >
                5PAPAPa
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/11-papapa"
            >
              <img
                alt="11PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490198/baixados_1_a01f56f752.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/11-papapa"
              >
                11PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas ornare mollis dui congue venenatis. 
            </p>
          </div>
          <div
            class="c16"
          >
            <a
              href="/post/1-papapa"
            >
              <img
                alt="1PAPAPA"
                class="c17"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489672/baixados_58294ddddf.jpg"
              />
            </a>
            <h2
              class="c5 c12"
            >
              <a
                href="/post/1-papapa"
              >
                1PAPAPA
              </a>
            </h2>
            <p
              class=""
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum.
            </p>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="c18"
    >
      <footer
        class="c19"
      >
        <div
          class="c20 c21"
        >
          texto do footer
        </div>
      </footer>
    </footer>
    <a
      aria-label="Go to top"
      class="c22"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
